models/source_list.py

```diff
@@ -53,10 +67,106 @@ class SourceList(Base, FileOnDiskMixin):
         )
     )
 
+    def __init__(self, *args, **kwargs):
+        FileOnDiskMixin.__init__(self, *args, **kwargs)
+        SeeChangeBase.__init__(self)  # don't pass kwargs as they could contain non-column key-values
+
+        self._data = None
+
+        # manually set all properties (columns or not)
+        for key, value in kwargs.items():
+            if hasattr(self, key):
+                setattr(self, key, value)
+
+    @orm.reconstructor
+    def init_on_load(self):
+        Base.init_on_load(self)
+        FileOnDiskMixin.init_on_load(self)
+        self._data = None
+
+    def __repr__(self):
+        output = (
+            f'<SourceList(id={self.id}, '
+            f'image_id={self.image_id}, '
+            f'is_sub={self.is_sub}), '
+            f'num_sources= {self.num_sources}>'
+        )
+
+        return output
+
+    @property
+    def data(self):
+        """
+        The data in this source list. A table of sources and their properties.
+        """
+        if self._data is None and self.filepath is not None:
+            self._data = self.load()
+        return self._data
+
+    @data.setter
+    def data(self, value):
+        # TODO: add astropy table?
+        if isinstance(value, pd.DataFrame):
+            value = value.to_records(index=False)
+
+        if not isinstance(value, np.ndarray) or value.dtype.names is None:
+            raise TypeError("data must be a pandas DataFrame or numpy recarray")
+
+        self._data = value
+        self.num_sources = len(value)
+
+    @property
+    def mjd(self):
```

Comment:
THis property shouldn't be here; you can get it with a join to the image table.

Reply:
That was a stupid comment. :D

My brain thought it was looking at a column definition.  Please ignore.

Reply:
You can also delete comments :)